docs/docs/user/references/components/overview.md

```diff
@@ -0,0 +1,32 @@
+# Overview 
+
+This section explains the different components of KPOps, 
+their usage and configuration in the pipeline 
+definition [`pipeline.yaml`](../../../resources/pipeline-components/pipeline).
+
+<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.4.0/css/all.min.css">
+
+```mermaid
+flowchart BT
+    PipelineComponent --> BaseDefaultsComponent
+    KubernetesApp --> PipelineComponent
+    KafkaConnector --> PipelineComponent
+    KafkaApp --> KubernetesApp
+    StreamsApp --> KafkaApp
+    ProducerApp --> KafkaApp
+    KafkaSourceConnector --> KafkaConnector
+    KafkaSinkConnector --> KafkaConnector
+    
+    click KubernetesApp "../kubernetes-app"
+    click KafkaApp "../kafka-app"
+    click StreamsApp "../streams-app"
+    click ProducerApp "../producer-app"
+    click KafkaSourceConnector "../kafka-source-connector"
+    click KafkaSinkConnector "../kafka-sink-connector"
```

Comment:
```suggestion
    click KafkaSinkConnector "../kafka-sink-connector"
    click KafkaConnector "../../defaults.md/#kafka-connector
```

Reply:
This is a bit confusing 🤔 Should we consider moving the kafka-connector description to the components too?

Reply:
Moved the KafkaConnector to the components section.

Reply:
I really like how you did it. :)